Objects/genobject.c

```diff
@@ -1771,6 +1771,7 @@ async_gen_asend_send(PyAsyncGenASend *o, PyObject *arg)
 
     if (o->ags_state == AWAITABLE_STATE_INIT) {
         if (o->ags_gen->ag_running_async) {
+            o->ags_state = AWAITABLE_STATE_CLOSED;
```

Comment:
this was set in `async_gen_athrow_send` but forgotten in `async_gen_asend_send`

Reply:
https://github.com/python/cpython/blob/8418dcc5d3190a199d2236e99e30f2c146939e95/Objects/genobject.c#L2098-L2113

Reply:
probably could do with a test

Reply:
test added in 1220aee3d7a1a8e72903abc815fdf96f31c5ce34